--- tomcat7.orig    2014-08-01 02:50:44.390633083 +0900
+++ tomcat7 2014-08-01 02:53:42.542636501 +0900
@@ -193,8 +193,8 @@
	fi

	log_daemon_msg "Starting $DESC" "$NAME"
-	if start-stop-daemon --test --start --pidfile "$CATALINA_PID" \
-		--user $TOMCAT7_USER --exec "$JAVA_HOME/bin/java" \
+	if ! start-stop-daemon --test --status --pidfile "$CATALINA_PID" \
+		--user $TOMCAT7_USER \
		>/dev/null; then

		# Regenerate POLICY_CACHE file
@@ -215,8 +215,8 @@

		catalina_sh start $SECURITY
		sleep 5
-        	if start-stop-daemon --test --start --pidfile "$CATALINA_PID" \
-			--user $TOMCAT7_USER --exec "$JAVA_HOME/bin/java" \
+		if ! start-stop-daemon --test --status --pidfile "$CATALINA_PID" \
+			--user $TOMCAT7_USER \
			>/dev/null; then
			if [ -f "$CATALINA_PID" ]; then
				rm -f "$CATALINA_PID"
@@ -247,18 +247,19 @@
		fi
		rm -f "$CATALINA_PID"
		rm -rf "$JVM_TMP"
+		log_end_msg 0
	else
		log_progress_msg "(not running)"
+		log_end_msg 1
	fi
-	log_end_msg 0
	set -e
	;;
    status)
	set +e
-	start-stop-daemon --test --start --pidfile "$CATALINA_PID" \
-		--user $TOMCAT7_USER --exec "$JAVA_HOME/bin/java" \
+	start-stop-daemon --test --status --pidfile "$CATALINA_PID" \
+		--user $TOMCAT7_USER \
		>/dev/null 2>&1
-	if [ "$?" = "0" ]; then
+	if [ "$?" != "0" ]; then

		if [ -f "$CATALINA_PID" ]; then
		    log_success_msg "$DESC is not running, but pid file exists."

